### DISCUSSION TITLE
Would be cool to be able to specify sheets instead of the first one

### DISCUSSION THREAD

**Discussion by spookyuser** [fureweb-com/public-google-sheets-parser#6] [Ideas]
Just want to say I love this library, and I'm happy to even make a PR attempt at this, so that you could have a psuedo database in sheets lol

**Comment by spookyuser** [comment]
Alright this is my attempt 🤠 #7

**Reply by spookyuser** [reply]
I made my own spreadsheet to test this feature so maybe the second sheet should be moved to the main test document, my spreadsheet is here if you want to copy it https://docs.google.com/spreadsheets/d/1wdU-Cf7Kn5ZvhcvRePOoL10urdTA6TWGndBSKDAiYbQ/edit#gid=0

**Reply by fureweb-com** [reply]
Thanks for suggesting a nice feature!
I will comment on the pull request. :)

**Comment by cheveedodd** [comment]
Would it be possible to implement this using the gid provided in sheets urls instead of the name of the sheet?
In @spookyuser example, the second sheet URL is:
https://docs.google.com/spreadsheets/d/1wdU-Cf7Kn5ZvhcvRePOoL10urdTA6TWGndBSKDAiYbQ/edit#gid=1528085346

In order to parse that correctly,  when constructing the URL you would use `gid` in place of `sheet`

`    
if (this.tab) {
    url = url.concat(`sheet=${this.tab}`)
}
`

**Reply by fureweb-com** [reply]
Hello, thank you for your proposal!

I have reviewed the content you provided, and I've considered using the gid instead of sheetName to provide a better user experience. If more users request supporting both sheetName and gid in the future, I will consider it when deciding on the next version.

Personally, I think there would not be much benefit in using gid compared to sheetName, as the numbers representing gid are not easily readable and may not guarantee readability in client code when trying to access the desired sheet data. However, I do understand that it could be a useful option for those who need it.

For now, I will wait for more requests regarding this matter before making a decision on whether to proceed with the change.

Thank you for understanding.

**Reply by cheveedodd** [reply]
We have our users copy/paste the URL of their sheet into our tool. It would be convenient for everyone involved if they didn't also need to remember/type in a sheet name in addition since the gid is already present in the URL and doesn't change if they alter the sheet name in the future.

My proposal could be implemented easily by having the second argument for your parser be an object that contains either sheetID or GID
{ sheetName: 'Sheet2' } or { sheetID: 1528085346 } and if it contains both defer to the sheetID since it doesn't change.

Thanks for considering!

**Reply by fureweb-com** [reply]
Thanks to @SoulAuctioneer for implementing the feature to support sheetId.

It is released as v1.3.0 and is available for use. Please check it out :) @cheveedodd

**Reply by cheveedodd** [reply]
I saw that this morning and immediately implemented it. Thanks so much!!

**Comment by SoulAuctioneer** [comment]
I've opened a PR with an update to support sheet ID:
https://github.com/fureweb-com/public-google-sheets-parser/pull/17
